java/SQSSink/src/main/java/com/amazonaws/services/msf/SQSStreamingJob.java

```diff
@@ -0,0 +1,84 @@
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+
+public class SQSStreamingJob {
+
+    // Name of the local JSON resource with the application properties in the same format as they are received from the Amazon Managed Service for Apache Flink runtime
+    private static final String LOCAL_APPLICATION_PROPERTIES_RESOURCE = "flink-application-properties-dev.json";
+
+    // Create ObjectMapper instance to serialise POJOs into JSONs
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static boolean isLocal(StreamExecutionEnvironment env) {
+        return env instanceof LocalStreamEnvironment;
+    }
+
+    /**
+     * Load application properties from Amazon Managed Service for Apache Flink runtime or from a local resource, when the environment is local
+     */
+    private static Map<String, Properties> loadApplicationProperties(StreamExecutionEnvironment env) throws IOException {
+        if (isLocal(env)) {
+            return KinesisAnalyticsRuntime.getApplicationProperties(
+                    SQSStreamingJob.class.getClassLoader()
+                            .getResource(LOCAL_APPLICATION_PROPERTIES_RESOURCE).getPath());
+        } else {
+            return KinesisAnalyticsRuntime.getApplicationProperties();
+        }
+    }
+
+    private static DataGeneratorSource<StockPrice> getStockPriceDataGeneratorSource() {
+        long recordPerSecond = 100;
+        return new DataGeneratorSource<>(
+                new StockPriceGeneratorFunction(),
+                Long.MAX_VALUE,
+                RateLimiterStrategy.perSecond(recordPerSecond),
+                TypeInformation.of(StockPrice.class));
+    }
+
+    private static SqsSink<String> createSQSSink(
+            Properties sinkProperties) {
+        return SqsSink.<String>builder()
+                .setSqsSinkElementConverter(SqsSinkElementConverter.<String>builder().build())
+                .setSqsUrl("https://sqs.us-east-1.amazonaws.com/012345678901/MyTestQueue")
+                .setSqsClientProperties(sinkProperties)
+                .build();
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        // set up the streaming execution environment
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final Map<String, Properties> applicationProperties = loadApplicationProperties(env);
+
+        // Source
+        DataGeneratorSource<StockPrice> source = getStockPriceDataGeneratorSource();
+
+        // DataStream from Source
+        DataStream<StockPrice> input = env.fromSource(
+                source, WatermarkStrategy.noWatermarks(), "data-generator").setParallelism(1);
+
+        SqsSink<String> sink = createSQSSink(applicationProperties.get("OutputStream0"));
+
+        input.map(StockPrice::toString).uid("object-to-string-map")
```

Comment:
If we change to send `StockPrice` directly, as mentioned above, the `map` becomes redundant.